apps/api/src/common/env/env.schema.ts

```diff
@@ -189,8 +190,17 @@ const prodSchema = z.object({
 
 export type EnvSchemaType = z.infer<typeof prodSchema>
 
-export const EnvSchema = z.discriminatedUnion('NODE_ENV', [
-  e2eEnvSchema,
-  prodSchema,
-  devSchema
-])
+export const EnvSchema = z
+  .discriminatedUnion('NODE_ENV', [e2eEnvSchema, prodSchema, devSchema])
+  .refine(
+    (data) => {
```

Comment:
isnt the right place to add it. i would suggest you to create a refinement object, and then use it individually for secret, variable and environment names. and not use it in here